nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/ui/canvas/canvas.component.ts

```diff
@@ -168,9 +169,9 @@ export class Canvas implements OnInit, OnDestroy {
 
         // handle bulk component selection
         this.store
-            .select(selectCurrentProcessGroupId)
+            .select(selectFlowLoadingStatus)
             .pipe(
-                filter((processGroupId) => processGroupId != initialState.id),
+                filter((status) => status === 'complete'),
```

Comment:
All three of the changes moving to `selectFlowLoadingStatus` will re-emit when the canvas refreshes. This can have unintended side effects like centering or issuing unnecessary actions. Following the `filter` here we could simply `take(1)` in order to not re-emit beyond this point.